### PR TITLE
 Raise KeyError for missing keys

### PR DESCRIPTION
**Pull request by noelevans**
Fixing issue neovim#402
"Raise KeyError for missing options / vars".

### CODE REVIEW

File: pynvim/api/common.py [hunk 1]
```diff
@@ -85,7 +85,10 @@ def __init__(self, obj, get_method, set_method=None):
 
     def __getitem__(self, key):
         """Return a map value by key."""
-        return self._get(key)
+        try:
+            return self._get(key)
+        except Exception:
+            raise KeyError
```
Comment (mhinz) [inline pynvim/api/common.py]: Is the generic `KeyError` the proper thing to throw for the API?

Maybe throw `NvimAPIKeyError` which derives from `KeyError`? Would be more precise and `exception KeyError` in user code would still work.

**Comment by bfredl** [comment]
> Maybe throw NvimAPIKeyError which derives from KeyError? 

We do not need to create a new type for every specific place in which a specific error can be thrown, this is just typical overengineering. You can already check the error for vars/options only just by using `try/except` around the specific access (or even better use `.get()` if you expect the key to be optional). 

Also see #405

**Comment by blueyed** [comment]
Yeah, that's a duplicate of https://github.com/neovim/pynvim/pull/405 unfortunately.

File: pynvim/api/common.py (hunk 1 above)
Comment (blueyed) [reply pynvim/api/common.py]: Had the same discussion in https://github.com/neovim/pynvim/pull/405#discussion_r315202545 already.

**Comment by blueyed** [comment]
@noelevans 
Thanks for the PR - please review #405 instead.

**Pull request closed without merging**